src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataDeserializerContext.cs

```diff
@@ -62,20 +66,75 @@ internal bool IsDeltaOfT
             {
                 if (!_isDeltaOfT.HasValue)
                 {
-                    _isDeltaOfT = ResourceType != null && TypeHelper.IsGenericType(ResourceType) && ResourceType.GetGenericTypeDefinition() == typeof(Delta<>);
+                    _isDeltaOfT = ResourceType != null && TypeHelper.IsGenericType(ResourceType) && (ResourceType.GetGenericTypeDefinition() == typeof(Delta<>) ||
+                        ResourceType.GetGenericTypeDefinition() == typeof(DeltaDeletedEntityObject<>));
                 }
 
                 return _isDeltaOfT.Value;
             }
         }
 
+        internal bool IsDeletedDeltaOfT
+        {
+            get
+            {
+                if (!_isDeletedDeltaOfT.HasValue)
+                {
+                    _isDeletedDeltaOfT = ResourceType != null && TypeHelper.IsGenericType(ResourceType) && (ResourceType.GetGenericTypeDefinition() == typeof(DeltaDeletedEntityObject<>) ||
+                        ResourceType.GetGenericTypeDefinition() == typeof(DeltaDeletedEntityObject<>));
+                }
+
+                return _isDeletedDeltaOfT.Value;
+            }
+        }
+
+        internal bool IsDeltaEntity
+        {
+            get
+            {
+                if (!_isDeltaEntity.HasValue)
+                {
+                    _isDeltaEntity = ResourceType != null && (ResourceType == typeof(EdmDeltaEntityObject) || ResourceType == typeof(EdmDeltaDeletedEntityObject));
+                }
+
+                return _isDeltaEntity.Value;
+            }
+        }
+
+        internal bool IsDeltaDeletedEntity
+        {
+            get
+            {
+                if (!_isDeltaDeletedEntity.HasValue)
+                {
+                    _isDeltaDeletedEntity = ResourceType != null && ResourceType == typeof(EdmDeltaDeletedEntityObject);
+                }
+
+                return _isDeltaDeletedEntity.Value;
+            }
+        }
+
+        internal bool IsChangedObjectCollection
+        {
+            get
+            {
+                if (!_isChangedObjectCollection.HasValue)
+                {
+                    _isChangedObjectCollection = ResourceType != null && (ResourceType == typeof(EdmChangedObjectCollection) || (TypeHelper.IsGenericType(ResourceType) &&
+                        ResourceType.GetGenericTypeDefinition() == typeof(DeltaSet<>)));
+                }
+
+                return _isChangedObjectCollection.Value;
+            }
+        }
+
         internal bool IsUntyped
         {
             get
             {
                 if (!_isUntyped.HasValue)
                 {
-                    _isUntyped = TypeHelper.IsTypeAssignableFrom(typeof(IEdmObject), ResourceType) ||
+                    _isUntyped = IsChangedObjectCollection ? !TypeHelper.IsGenericType(ResourceType) : (TypeHelper.IsTypeAssignableFrom(typeof(IEdmObject), ResourceType) && !IsDeltaOfT) ||
```

Comment:
Not sure I'm able to wrap my head around the logic here

Reply:
it checks whether the `ResourceType` is untyped or not.  If it is a `ChangedObjectCollection` then it should not be a generic type if it untyped etc etc.